### PR TITLE
Implement RestoreObject API

### PR DESCRIPTION
**Pull request by vadmeste**
Implement the following API https://docs.aws.amazon.com/AmazonS3/latest/API/API_RestoreObject.html

### CODE REVIEW

**Review by klauspost — APPROVED** [review]
LGTM, just a minor change.

**Comment by vadmeste** [comment]
@harshavardhana @klauspost I added a separate commit which parses x-amz-restore header in a HEAD request. PTAL again.

File: api-datatypes.go [hunk 1]
```diff
@@ -85,6 +86,12 @@ type UploadInfo struct {
 	ExpirationRuleID string
 }
 
+// RestoreInfo contains information of the restore operation of an archived object
+type RestoreInfo struct {
+	OngoingRestore bool
+	ExpiryTime     time.Time
```
Comment (klauspost) [inline api-datatypes.go:92]: It is not obvious what "ExpiryTime" is. Add comments to explain fields.

Do we know the start time of the restore. Would be nice to have, if feasible.

Comment (vadmeste) [reply api-datatypes.go]: > It is not obvious what "ExpiryTime" is. Add comments to explain fields.

Done. Restore API will actually get a copy of the archived object in the first time. The object will be automatically removed from the fast tier after x days.

> Do we know the start time of the restore. Would be nice to have, if feasible.

No, S3 doesn't expose that.

**Review by klauspost — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**